### DISCUSSION TITLE
Benefit of the adding of the `.bin` folders to the PATH to run a command/script manually

### DISCUSSION THREAD

**Discussion by ghost** [xpack/xpm-js#193] [Q&A] [answered]
Hi,

I would like to know if it is possible to benefit of the adding of the `.bin` folders to the PATH? (First stage of the `xpm run`, cf. [doc](https://xpack.github.io/xpm/cli/run/#path)).

My goal is to run a script, or command without write it as an action into the _package.json_ file.
For instance:

```pwsh
meson compile my_target
```

In advance, thank you.

**Comment by ilg-ul** [comment]
The purpose of `xpm run` is exactly to automate things, like path management.

If you prefer to run commands directly, sure, but you must manually set the PATH to your desired location before invoking the commands.

**Reply by ghost** [reply]
Is it possible to execute (on the fly) something like this: `xpm run meson compile my_target`?

(Without write an action)

**Reply by ilg-ul** [reply, marked as answer]
`xpm run` requires an action name defined in package.json.

**Reply by ghost** [reply]
Thank you Liviu!
